ckan/lib/search/common.py

```diff
@@ -48,23 +51,48 @@ def is_available():
     """
     try:
         conn = make_connection()
-        conn.query("*:*", rows=1)
+        conn.search(q="*:*", rows=1)
     except Exception, e:
         log.exception(e)
         return False
-    finally:
-        if 'conn' in dir():
-            conn.close()
-
     return True
 
 
-def make_connection():
-    from solr import SolrConnection
+def make_connection(decode_dates=True):
     solr_url, solr_user, solr_password = SolrSettings.get()
-    assert solr_url is not None
-    if solr_user is not None and solr_password is not None:
-        return SolrConnection(solr_url, http_user=solr_user,
-                              http_pass=solr_password)
+    if decode_dates:
```

Comment:
If I recall correctly

* solrpy requests xml from solr and parses and  allows you to search in two different ways
`search` (returns datetime types) and `raw_search` (returns text)
* pysolr parses the json (which does not provide a datetime type) and returns that in its search results

Since this is ckan, we like being difficult so we obviously use both raw and non raw searches in solrpy in differeent places and expect the results from solr to be a datetime and text other times. 

So I had to add the decoding of dates to pysolr for the functions expecting datetime objects and some for when it wasn't

if i recall I lifted the datetime regex from pysolr (https://github.com/toastdriven/pysolr/blob/master/pysolr.py#L68) I think I should probably import it instead.

I think overall, the real fix would be fix the datetime usage across all of ckan so whenever you are working with a date it is a datetime object, but I didn't think that would be realistic.